api/scpca_portal/test/views/test_dataset.py

```diff
@@ -88,3 +90,79 @@ def test_delete_is_not_allowed(self):
 
         self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    @patch("scpca_portal.models.Job.submit")
+    def test_create_submit_job(self, mock_submit_job):
+        url = reverse("datasets-list", args=[])
+
+        # Assert job not started
+        data = {
+            "data": DatasetCustomSingleCellExperiment.VALUES.get("data"),
+            "email": DatasetCustomSingleCellExperiment.VALUES.get("email"),
+            "format": DatasetCustomSingleCellExperiment.VALUES.get("format"),
+            "start": False,
+        }
+        response = self.client.post(url, data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        mock_submit_job.assert_not_called()
+
+        # Assert job started
+        data = {
+            "data": DatasetCustomSingleCellExperiment.VALUES.get("data"),
+            "email": DatasetCustomSingleCellExperiment.VALUES.get("email"),
+            "format": DatasetCustomSingleCellExperiment.VALUES.get("format"),
+            "start": True,
+        }
+        response = self.client.post(url, data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        mock_submit_job.assert_called_once()
+
+        # Assert that start attribute was reset to False after kicking off job
+        created_dataset_id = response.json().get("id")
+        created_dataset = Dataset.objects.filter(id=created_dataset_id).first()
+        self.assertFalse(created_dataset.start)
+
+    @patch("scpca_portal.models.Job.submit")
+    def test_update_submit_job(self, mock_submit_job):
+        dataset = Dataset(
+            data=DatasetCustomSingleCellExperiment.VALUES.get("data"),
+            email=DatasetCustomSingleCellExperiment.VALUES.get("email"),
+            format=DatasetCustomSingleCellExperiment.VALUES.get("format"),
+            start=False,
+        )
+        url = reverse("datasets-detail", args=[dataset.id])
+
+        # Assert job not started when is_started is True
+        dataset.is_started = True
+        dataset.save()
+        data = {
+            "start": True,
+        }
+        response = self.client.put(url, data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        mock_submit_job.assert_not_called()
+        dataset.is_started = False
+        dataset.save()
+
+        # Assert job not started when is_processing is True
+        dataset.is_processing = True
+        dataset.save()
+        data = {
+            "start": True,
+        }
+        response = self.client.put(url, data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        mock_submit_job.assert_not_called()
+        dataset.is_processing = False
+        dataset.save()
+
+        # Assert job started
+        data = {
+            "data": DatasetCustomSingleCellExperiment.VALUES.get("data"),
+            "start": True,
+        }
+        response = self.client.post(url, data)
```

Comment:
Please replace the `post` method with `put` to prevent `HTTP_405_METHOD_NOT_ALLOWED`.

Reply:
We generally dont support partial updates so only post is ok.